provider/k8s/process.go

```diff
@@ -428,12 +428,13 @@ func (p *Provider) podSpecFromService(app, service, release string) (*ac.PodSpec
 			c.Image = fmt.Sprintf("%s:%s.%s", repo, service, r.Build)
 
 			for _, r := range s.ResourceMap() {
+				key := strings.Join(strings.Split(r.Env, "_")[1:], "_")
```

Comment:
@Twsouza This breaks things for any resource with a 'multi-part' name.
i.e. I have a resource called `primary-database` in the convox.yml.  This means I end up with a env var called `PRIMARY_DATABASE_URL`.  
But splitting on the underscore and pulling out everything after the first part, means that the `key` is now `DATABASE_URL` which doesn't exist in the ConfigMap so deployments fail as that can't be found.

Reply:
hey @beastawakens thanks for pointing this out. I've opened a PR to fix it, see https://github.com/convox/convox/pull/517